src/System.Collections/tests/Performance/Perf.HashSet.cs

```diff
@@ -0,0 +1,501 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+/// <summary>
+/// Performs quick perf test of HashSet operations. Tests with T = int
+/// 
+/// This is intended for the regular perf runs to catch perf regressions. 
+/// Use HashSetDetailedTests for detailed perf analysis.
+/// </summary>
+public class Perf_HashSet
+{
+    private const int InitialSetSize = 8000000;
+    private const int InitialSetSize_small = 320000;
+    private const int MaxStartSize = 32000;
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Add(int initialSetSize, int countToAdd)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            // n is varying start size of set
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] stuffToAdd = intGenerator.MakeNewTs(countToAdd);
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+            {
+                foreach (int thing in stuffToAdd)
+                {
+                    theSet.Add(thing);
+                }
+            }
+        }
+    }
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Contains(int initialSetSize, int countToCheck)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] subsetToCheck = intGenerator.GenerateSelectionSubset(startingElements, countToCheck);
```

Comment:
Same question (I'll stop asking it about later tests)... can the above be moved out of the loop?